src/com/group6/users/UserStorage.java

```diff
@@ -0,0 +1,51 @@
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UserStorage {
+    public static Map<String, User> loadUsers(){
+        Map<String, User> users = new HashMap<>();
+        loadUsersFromFile(users, "applicants.txt", UserRole.APPLICANT);
+        loadUsersFromFile(users, "officers.txt", UserRole.OFFICER);
+        loadUsersFromFile(users, "managers.txt", UserRole.MANAGER);
+        return users;
+    }
+
+    public static void loadUsersFromFile(Map<String, User> users, String filename, UserRole role){
+        try(BufferedReader reader = new BufferedReader(new FileReader(filename))){
+            String line;
+            while ((line = reader.readLine()) != null){
+                String[] parts = line.split(",");
+                if (parts.length == 5){
+                    User user = createUserByRole(parts, role);
+                    users.put(user.getNric(), user);
+                }
+            }
+        } catch (IOException e){
+            System.out.println("Error reading from file " + filename);
+        }
+    }
+
+    private static User createUserByRole(String[] parts, UserRole role){
+        return switch (role){
+            case APPLICANT -> new Applicant(parts[0], parts[1], Integer.parseInt(parts[2]), parts[3], parts[4]);
+            case OFFICER -> new HDBOfficer(parts[0], parts[1], Integer.parseInt(parts[2]), parts[3], parts[4]);
+            case MANAGER -> new HDBManager(parts[0], parts[1], Integer.parseInt(parts[2]), parts[3], parts[4]);
+        };
+    }
+
+    public static void saveUser(User user){
+        String filename = switch (user.getRole()){
+            case APPLICANT -> "applicants.txt";
+            case OFFICER -> "officers.txt";
+            case MANAGER -> "managers.txt";
+        };

```

Comment:
So for smth like this, assuming you do remove `getRole`, you can consider using 
`if (user instanceof HDBOfficer) {}` // .... 
But ofc, there is a risk if you accidentally order it like:
```
 if (user instanceof Applicant) {}
 else if (user instanceof HDBOfficer) {} // This will never be hit.
 ```
 
Your `getRole()` approach does solve this problem. 

Alternatively:
You can use `user.getClass()` and compare it with the `class`. 
Example:
```
Class<?> clazz = user.getClass();
switch (clazz) {
  case Applicant.class -> // ...
  case HDBOfficer.class -> // ...
  case HDBManager.class -> // ...
}
```

I leave it to you on whats the best course of action for this.